File: docker-python/cgi-bin/registration.py
#!/usr/bin/python3.9
print('Content-Type: text/html\n')
"""
データをmysqlに登録するプログラム
"""

import MySQLdb
import sys
import hashlib
import json
a = sys.stdin.readline()
receive = json.loads(a)
receive["passward"] = hashlib.sha256(str(receive["passward"]).encode()).hexdigest()
connection = MySQLdb.connect(
    host='localhost',
    user='root',
    passwd='',
    db='onlinetest',
    charset='utf8'
    )
cursor = connection.cursor()
b = cursor.execute("select exists(select id from users where id = %s)", [receive["id"]])
if cursor.fetchone()[0] == 0:
	cursor.execute("insert into users (id, name, passward, question, answer) values (%(id)s, %(username)s, %(passward)s, %(question)s, %(answer)s)", receive)
	print(1)
#print("insert into users (id, username, passward, question, answer) values (\"" + str(receive["id"]) + "\",\"" + str(receive["username"]) + "\",\"" + hashlib.sha256(str(receive["passward"]).encode()).hexdigest() + "\",\"" + str(receive["question"]) + "\",\"" + hashlib.sha256(str(receive["answer"]).encode()).hexdigest() + "\",)")
else:
	print(0)

connection.commit()																				#connectionを保存する
connection.close()																				#connectionをクローズ

File: docker-laravel/backend/public/php/registration.php
<?php
/*
引数  id, name, passward, question, answer
受けとったデータのidがもしデータベースに存在しないならpasswardをハッシュ化してデータを書き込む
そうでない場合は0を返却し存在することを伝える
また、異常終了時は3を返却する。

*/
try {
$pdo = new PDO("mysql:dbname=laravel_local;host=db;charset=utf8mb4", "phper", "secret", 
		[
            PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
            PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
        ]
	);
$pass = $_POST["passward"];
#$pass = "1";
$passward = hash("sha256",$pass);
$id = $_POST["id"];
$stmt = $pdo->prepare("select exists(select id from users where id = ? )");
$stmt->bindValue(1, (int)$id, PDO::PARAM_INT);
$stmt->execute();
$a = $stmt->fetch(PDO::PARAM_INT);

if($a[0] == 0){
	$stmt = $pdo-> prepare("insert into users (id, name, passward, question, answer) values (?, ?, ?, ?, ?)");
	$stmt->bindValue(1, (int)$id, PDO::PARAM_INT);
	$stmt->bindValue(2, $_POST["username"], PDO::PARAM_STR);
	$stmt->bindValue(3, $passward, PDO::PARAM_STR);
	$stmt->bindValue(4, $_POST["question"], PDO::PARAM_STR);
	$stmt->bindValue(5, $_POST["answer"], PDO::PARAM_STR);
	$stmt->execute();
	echo 1;
}
else
	echo 0;


	} catch (PDOException $e) {
        //echo $e->getMessage();
        echo -1;
        exit;
    }

    ?>